src/AstTransforms.cpp

```diff
@@ -404,9 +404,9 @@ void ResolveAliasesTransformer::removeComplexTermsInAtoms(AstClause& clause) {
 
     // add variable constraints to clause
     for (const auto& cur : map) {
-        clause.addToBody(std::unique_ptr<AstLiteral>(
-                new AstConstraint(BinaryConstraintOp::EQ, std::unique_ptr<AstArgument>(cur.second->clone()),
-                        std::unique_ptr<AstArgument>(cur.first->clone()))));
+        clause.addToBody(std::unique_ptr<AstLiteral>(new AstBinaryConstraint(BinaryConstraintOp::EQ,
```

Comment:
Could this not be `std::make_unique<AstBinaryConstraint>(...)`?

Reply:
True and false identifiers are standard in most languages.
I feel ok to sacrifice existing true/false identifiers for I/O since there are not so important for expressing the semantics of a logical program. I suggest to rename existing I/O identifiers from true/false to on/off.